vochain/account.go

```diff
@@ -208,251 +223,248 @@ func (v *State) IncrementAccountProcessIndex(accountAddress common.Address) erro
+	costSetAccountInfoURI, err := state.TxCost(models.TxType_SET_ACCOUNT_INFO_URI, false)
 	if err != nil {
-		return nil, fmt.Errorf("cannot check if account %s exists: %w", tx.Account, err)
+		return fmt.Errorf("cannot get tx cost: %w", err)
 	}
-	if acc == nil {
-		returnValues.Account = returnValues.TxSender
+	// check tx sender balance
+	if txSenderAccount.Balance < costSetAccountInfoURI {
+		return fmt.Errorf("unauthorized: %s", ErrNotEnoughBalance)
 	}
-	// if not exist create new one
-	if txSender == nil {
-		returnValues.CreateAccount = true
-		if tx.FaucetPackage == nil {
-			return returnValues, nil
-		}
-		if len(tx.FaucetPackage.Payload) == 0 {
-			return nil, fmt.Errorf("invalid faucet package payload")
-		}
-		faucetPayload := &models.FaucetPayload{}
-		if err := proto.Unmarshal(tx.FaucetPackage.Payload, faucetPayload); err != nil {
-			return nil, fmt.Errorf("could not unmarshal faucet package: %w", err)
-		}
-		returnValues.CreateAccountWithFaucet = true
-		if !bytes.Equal(faucetPayload.To, returnValues.TxSender.Bytes()) {
-			return nil, fmt.Errorf("payload to and tx sender missmatch (%x != %x)",
-				faucetPayload.To, returnValues.TxSender.Bytes())
-		}
-		// get issuer address from faucetPayload
-		issuerAddress, err := ethereum.AddrFromSignature(tx.FaucetPackage.Payload, tx.FaucetPackage.Signature)
-		if err != nil {
-			return nil, err
-		}
-		// check issuer nonce not used
-		b := make([]byte, 8)
-		binary.LittleEndian.PutUint64(b, faucetPayload.Identifier)
-		key := issuerAddress.Bytes()
-		key = append(key, b...)
-		used, err := state.FaucetNonce(crypto.Sha256(key), false)
-		if err != nil {
-			return nil, fmt.Errorf("cannot check faucet nonce: %w", err)
-		}
-		if used {
-			return nil, fmt.Errorf("faucet package identifier %d already used", faucetPayload.Identifier)
-		}
-		// check issuer have enough funds
-		issuerAcc, err := state.GetAccount(issuerAddress, false)
-		if err != nil {
-			return nil, fmt.Errorf("cannot get faucet account: %w", err)
-		}
-		if issuerAcc == nil {
-			return nil, fmt.Errorf("the account signing the faucet payload does not exist (%x)", issuerAddress)
-		}
-		cost, err := state.TxCost(models.TxType_COLLECT_FAUCET, false)
-		if err != nil {
-			return nil, fmt.Errorf("cannot get %s tx cost: %w", models.TxType_COLLECT_FAUCET, err)
-		}
-		if issuerAcc.Balance < faucetPayload.Amount+cost {
-			return nil, fmt.Errorf("faucet does not have enough balance %d < %d", issuerAcc.Balance, faucetPayload.Amount+cost)
-		}
-		returnValues.FaucetPayloadSigner = issuerAddress
-		return returnValues, nil
+	// check info URI
+	infoURI := tx.GetInfoURI()
+	if len(infoURI) == 0 || len(infoURI) > types.MaxURLLength {
+		return fmt.Errorf("invalid URI, cannot be empty")
 	}
-	// check if delegate
-	if returnValues.TxSender != returnValues.Account {
-		if !acc.IsDelegate(returnValues.TxSender) {
-			return nil, fmt.Errorf("tx sender is not a delegate")
+	if txSenderAddress == txAccountAddress {
```

Comment:
How can txSenderAddress and txAccountAddress be different? One is obtained from the other.

Reply:
if delegates can be different, the txSenderAddress is the address sending the tx, the txAccountAddress is the address from the Account field if the tx if not empty, otherwise is the same as txSenderAddress